### PR TITLE
7.get/api/reviews/review id comment counter

### PR DESCRIPTION
**Pull request by mini-wizbit**

### CODE REVIEW

**Comment by mini-wizbit** [comment]
Start on the 'add first test with happy path'

File: Model/review.model.js [hunk 1]
```diff
@@ -10,6 +10,16 @@ exports.selectReviewsById = (review_id) => {
...
         }
+      })
+      .then((reviewById) => {
+        const id = reviewById.review_id;
+        return db
+          .query("SELECT * FROM comments WHERE review_id = $1;", [id])
+          .then((results) => {
+            const numberOfComments = results.rows.length;
+            reviewById.comment_count = numberOfComments;
+            return reviewById;
```
Comment (emilyb93) [inline Model/review.model.js]: I would prefer that this is done in SQL using the COUNT() aggregate function with a join. Saves us doing two queries everytime and SQL is very good at these operations. Have a look at the notes for SQL relationships 
https://notes.northcoders.com/courses/js-back-end/sql-relationships

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Refactor into a single SQL query please, also dont add ALL of your files to a ticket. The patch and post files are not relevant to this ticket

File: Model/review.model.js [hunk 2]
```diff
@@ -10,6 +10,19 @@ exports.selectReviewsById = (review_id) => {
...
+        return db
+          .query(
+            "SELECT COUNT(*) FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id WHERE reviews.review_id = $1;",
+            [id]
+          )
+          .then((results) => {
+            const countNumber = parseInt(results.rows[0].count);
+            reviewById.comment_count = countNumber;
+            return reviewById;
+          });
```
Comment (emilyb93) [inline Model/review.model.js]: I was more alluding to use doing this whole model with a single query, we can actually combine our query to select all the columns from the reviews table, and then ALSO use our COUNT() in the same query! 
When we use SELECT we can give that as many arguements as we want like

```SELECT reviews.review_id, reviews.votes, COUNT() AS comment_count etc ```

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Would like to see this done as one query instead of 2

**Review by pkaodev — APPROVED** [review]
Nice one!  Merge away

**Pull request merged**